routes/vote/vote.controller.js

```diff
@@ -0,0 +1,53 @@
+const { checkSchema } = require('express-validator')
+const { doRedirect, renderWithData, checkErrors, returnToCheckAnswers } = require('./../../utils')
+const {
+  optInSchema,
+  confirmRegistrationSchema,
+} = require('./../../schemas')
+
+module.exports = function(app) {
+  app.get('/vote/optIn', renderWithData('vote/optIn'))
+  app.post(
+    '/vote/optIn',
+    checkSchema(optInSchema),
+    checkErrors('vote/optIn'),
+    postOptIn,
+    doRedirect,
+  )
+  app.get('/vote/confirmRegistration', renderWithData('vote/confirmRegistration'))
+  app.post(
+    '/vote/confirmRegistration',
+    checkSchema(confirmRegistrationSchema),
+    checkErrors('vote/confirmRegistration'),
+    postConfirmRegistration,
+    doRedirect,
+  )
+}
+
+const postOptIn = (req, res, next) => {
```

Comment:
Hmm, we have a [`doYesNo`](https://github.com/cds-snc/cra-claim-tax-benefits/blob/master/utils/index.js#L158) middleware that probably works for this use case, (except for the `voterPageEdited` thing, might require changing the key or some code). 

The [commit](https://github.com/cds-snc/cra-claim-tax-benefits/commit/0a48462236c58a46bc1f308be8a9735e062a1bbe) might help you, have a look.

I'd like to keep the custom Post methods to a minimum.

Reply:
doYesNo seems to specifically target deductions

```req.session.deductions[claim] = false

    // fields is an array
    fields.map(field => {
      // if the field is truthy
      if (req.session.deductions[field]) {
        // "Amount" fields are zeroed -- others are nulled
        if (field.endsWith('Amount')) {
          if (Object.keys(req.session.deductions[field]).includes('amount')) {
            req.session.deductions[field].amount = 0.0
          } else {
            req.session.deductions[field] = 0
          }
        } else {
          req.session.deductions[field] = null
        }
      }
    })```

Which is why I wrote a new one. We could look into refactoring?

Reply:
Yeah, we should refactor.

We could do `field.endsWith('Amount') || field.endsWith('Edited')`, since it sets it to `0`.

Reply:
Oh, except the `deductions` problem. 

Okay, let's leave this as-is for now.